public/graph.js

```diff
@@ -199,19 +199,17 @@ class GraphController {
                 .style("text-anchor", "middle")
                 .text("No. of Failed Message (s)");
 
-        // custom color scheme
-        let color_scheme = [
-                "#31cece",
-                "#f58231",
-                "#3cb44b",
-                "#CCCC00",
-                "#4363d8",
-                "#800000",
-                "#f032e6",
-                "#911eb4",
-                "#e6194B"
-            ],
-            color = d3.scaleOrdinal(color_scheme),
+        let color_scheme = [],
+            operators_with_color_identity = Object.keys(MNOColors);
+
+        // Generate color scheme based on operators identity
+        operators.forEach((operator, index) => {
+            if (operators_with_color_identity.includes(operator)) {
+                color_scheme[index] = MNOColors[operator];
+            }
+        });
+
+        let color = d3.scaleOrdinal(color_scheme),
             colorReceived = d3.scaleOrdinal(color_scheme).domain(receivedKeys),
             colorSent = d3.scaleOrdinal(color_scheme).domain(sentKeys),
             colorFailed = d3.scaleOrdinal(["#a82e2e"]).domain(["total_errored"]);
```

Comment:
Have realized this is close enough with somtel we should probably use #ff0000
```suggestion
            colorFailed = d3.scaleOrdinal(["#ff0000"]).domain(["total_errored"]);
```